dataflow/src/main/java/org/checkerframework/dataflow/busyexpression/BusyExprStore.java

```diff
@@ -0,0 +1,152 @@
+public class BusyExprStore implements Store<BusyExprStore> {
+
+    /** A set of busy expression abstract values. */
+    private final Set<BusyExprValue> busyExprValueSet;
+
+    /**
+     * Create a new BusyExprStore.
+     *
+     * @param busyExprValueSet a set of busy expression abstract values
+     */
+    public BusyExprStore(Set<BusyExprValue> busyExprValueSet) {
+        this.busyExprValueSet = busyExprValueSet;
+    }
+
+    /** Create a new BusyExprStore. */
+    public BusyExprStore() {
+        busyExprValueSet = new LinkedHashSet<>();
+    }
+
+    /**
+     * Kill expressions if they contain variable var.
+     *
+     * @param var a variable
+     */
+    public void killBusyExpr(Node var) {
+        Iterator<BusyExprValue> iter = busyExprValueSet.iterator();
+        while (iter.hasNext()) {
+            BusyExprValue busyExprValue = iter.next();
+            Node expr = busyExprValue.busyExpression;
+            if (isExprContainVariable(expr, var)) {
+                iter.remove();
+            }
+        }
+    }
+
+    /**
+     * Return if the expression contains variable var.
+     *
+     * @param expr the expression checked
+     * @param var the variable
+     * @return true if the expression contains the variable
+     */
+    public boolean isExprContainVariable(Node expr, Node var) {
+        if (expr instanceof BinaryOperationNode) {
+            BinaryOperationNode binaryNode = (BinaryOperationNode) expr;
+            return isExprContainVariable(binaryNode.getLeftOperand(), var)
+                    || isExprContainVariable(binaryNode.getRightOperand(), var);
+        }
+
+        return expr.equals(var);
+    }
+
+    /**
+     * Add busy expression e to busy expression value set.
+     *
+     * @param e the busy expression to be added
+     */
+    public void putBusyExpr(BusyExprValue e) {
+        busyExprValueSet.add(e);
+    }
+
+    /**
+     * Add expressions to the store, add sub-expressions to the store recursively
+     *
+     * @param e the expression to be added
+     */
+    public void addUseInExpression(Node e) {
+        if (e instanceof BinaryOperationNode) {
+            BusyExprValue busyExprValue = new BusyExprValue(e);
+            putBusyExpr(busyExprValue);
+            // recursively add expressions
+            BinaryOperationNode binaryNode = (BinaryOperationNode) busyExprValue.busyExpression;
+            addUseInExpression(binaryNode.getLeftOperand());
+            addUseInExpression(binaryNode.getRightOperand());
+        }
+    }
+
+    @Override
+    public BusyExprStore copy() {
+        return new BusyExprStore(new HashSet<>(busyExprValueSet));
+    }
+
+    @Override
+    public BusyExprStore leastUpperBound(BusyExprStore other) {
+        Set<BusyExprValue> busyExprValueSetLub = new HashSet<>(this.busyExprValueSet);
+        busyExprValueSetLub.retainAll(other.busyExprValueSet);
+
+        return new BusyExprStore(busyExprValueSetLub);
+    }
+    // flow graph / hierarchy
```

Comment:
Do we still need these comments?

Reply:
ahhh, that's a mistake. Thanks for pointing out！